### PR TITLE
SDK-134 Allow setting values in openmrs-runtime.properties

### PR DESCRIPTION
**Pull request by AdamGrzybkowski**
@rkorytkowski 
I've changed the property options a little bit. You can specify three values in distro.properties:
1) property.%propertyName%=value
2) property.%propertyName%.prompt=value
3) property.%propertyName%.default=value
With option 1 specified wizard won't prompt for this value, if there will be only option 2 wizard will ask for value with specified prompt(option 3 is optional for default value)

The properties are then saved in openmrs-server.properties as %propertyName%=value.

The property.%propertyName% form distro can be override be passing -DpropertyName=value in command line.

Unfortunately putting properties in openmrs-servers.properties and setting them as system properties doesn't seem to solve @mogoodrich problem, I guess pih.config is not read as system value. I managed to setup PIH only with pih.config in runtime.properties. 

 Lastly I've changed promptForValueIfMissingWithDefault() method in wizzard, so the user does not have to write(in option nr 2, mentioned above) something like "Pleas specify value for someProperty, default: %s". The default String will be added automatically if defValue is specified.

### CODE REVIEW

**Comment by mogoodrich** [comment]
Thanks @AdamGrzybkowski.  So everything in openmrs-server.properties gets set as system properties on startup?

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/model/DistroProperties.java [hunk 1]
```diff
@@ -191,6 +194,42 @@ public String getName(){
...
+        return propertiesNames;
+    }
+
+    private String extractPropertyName(String key) {
+        int beginIndex = key.indexOf(".");
+        int endIndex = key.indexOf(".prompt");
+        if(endIndex != -1){
+            return key.substring(beginIndex + 1, endIndex);
+        } else  {
+            endIndex = key.indexOf(".default");
```
Comment (rkorytkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/model/DistroProperties.java]: .default should be as last so it should be rather key = StringUtils.stripStart(key, "property."); if (key.endsWith(".default")) { key = StringUtils.stripEnd(key, ".default")  } else { key = StringUtils.stripEnd(key, ".prompt"); }

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/model/Server.java [hunk 2]
```diff
@@ -712,6 +712,10 @@ public void setContainerId(String containerId) {
         setParam(PROPERTY_DOCKER_MYSQL, containerId);
     }
 
+    public void setPropertyValue(String propertyName, String value){
+        setParam("property."+propertyName+".default", value);
```
Comment (rkorytkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/model/Server.java]: Shouldn't you be calling setParam("property." + propertyName, value) to be concise with distro.properties?

**Comment by rkorytkowski** [comment]
@mogoodrich, to be precise every property starting from "property." gets set as a system property.

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/model/DistroProperties.java (hunk 1 above)
Comment (AdamGrzybkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/model/DistroProperties.java]: I think StringUtils.stripStart is not the best solution for this. If you try to do StringUtils.stripStart("property.pih.config", "property.") as a result you will get ih.config. 

Secondly I don't think the order is important here ? We just want all the property names, no matter if it is prompt or default

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/model/Server.java (hunk 2 above)
Comment (AdamGrzybkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/model/Server.java]: It definitely should, my mistake

File: maven-plugin/src/main/java/org/openmrs/maven/plugins/model/DistroProperties.java (hunk 1 above)
Comment (rkorytkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/model/DistroProperties.java]: My point is that you can have property.pih.default.config.default and your code breaks.
Not sure why stripStart would cut off 'p'... Seems like a bug or it treats '.' in a special way...

Comment (AdamGrzybkowski) [inline maven-plugin/src/main/java/org/openmrs/maven/plugins/model/DistroProperties.java]: According to docs [https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#stripStart(java.lang.String,%20java.lang.String)](StringUtils.stripStart) it works as expected. It deletes every char from the given string, not the whole one if present. 

I have changed the code and added your example to distro properties test file.

**Pull request merged**